### PR TITLE
Make search_string configurable for filtered_header_rule

### PR DESCRIPTION
**Pull request by neilharris123**
Currently the `search_string` for `filtered_header_rules` must have the same value as the rule name. If the search string is a header value that is sensitive, the only way with the existing module to add it as the search string, is to use the name of the rule. I don't believe having the same value for rule name and and search string makes sense in most cases. 
This change preserves the previous behaviour, but allows user to set their own search_string.

### CODE REVIEW

**Review by rpdelaney — CHANGES_REQUESTED** [review]
Thank you for your contribution! I left a comment.

File: variables.tf [hunk 1]
```diff
@@ -100,10 +100,11 @@ variable "filtered_header_rule" {
...
   default = {
-    header_types = []
-    priority     = 1
-    header_value = ""
-    action       = "block"
+    header_types  = []
+    priority      = 1
+    header_value  = ""
+    action        = "block"
+    search_string = string
```
[the remark this answers is not in the document]
Comment (rpdelaney) [reply variables.tf:107]: It may be more idiomatic to default to `nil`, actually. I'm not sure.

[the remark this answers is not in the document]
Comment (rpdelaney) [reply variables.tf:107]: @esacteksab What do you think? If you agree I think we could fix this up and merge it.

[the remark this answers is not in the document]
Comment (esacteksab) [reply variables.tf:107]: Let's do it!

[the remark this answers is not in the document]
Comment (rpdelaney) [reply variables.tf]: How's this? 444b5d3 @esacteksab

[the remark this answers is not in the document]
Comment (esacteksab) [reply variables.tf]: LGTM

**Review by rpdelaney — APPROVED** [review]
per Barry

**Comment by rpdelaney** [comment]
Shipped in `v2.6.0`

**Pull request merged**